src/components/calendar/bl-calendar.ts

```diff
@@ -0,0 +1,545 @@
+    }
+  }
+  clearRangePickerStyles() {
+    this.shadowRoot?.querySelectorAll(".range-day").forEach(day => {
+      day.classList.remove("range-day");
+    });
+    this.shadowRoot?.querySelectorAll(".range-start-day").forEach(day => {
+      day.classList.remove("range-start-day");
+    });
+    this.shadowRoot?.querySelectorAll(".range-end-day").forEach(day => {
+      day.classList.remove("range-end-day");
+    });
+  }
+  handleDate(date: CalendarDate) {
+    if (this.type !== CALENDAR_TYPES.range) {
+      if (date.getMonth() < this._calendarMonth) {
+        this.setPreviousCalendarView();
+      } else if (date.getMonth() > this._calendarMonth) {
+        this.setNextCalendarView();
+      }
+    }
+
+    if (this.type === CALENDAR_TYPES.single) {
+      this.handleSingleSelectCalendar(date);
+    } else if (this.type === CALENDAR_TYPES.multiple) {
+      this.handleMultipleSelectCalendar(date);
+    } else if (this.type === CALENDAR_TYPES.range) {
+      this.handleRangeSelectCalendar(date);
+    }
+
+    this._onBlCalendarChange(this._selectedDates);
+    this.requestUpdate();
+  }
+  handleSingleSelectCalendar(calendarDate: CalendarDate) {
+    this._selectedDates.splice(0, 1);
+    this._selectedDates.push(calendarDate);
+  }
+  handleMultipleSelectCalendar(calendarDate: CalendarDate) {
+    const dateExist = this._selectedDates.find(function (selectedDate) {
+      return selectedDate.getTime() === calendarDate.getTime();
+    });
+
+    if (dateExist) this._selectedDates.splice(this._selectedDates.indexOf(calendarDate), 1);
+    else this._selectedDates.push(calendarDate);
+  }
+  handleRangeSelectCalendar(calendarDate: CalendarDate) {
+    if (!this._selectedRangeDates.startDate) {
+      this._selectedRangeDates.startDate = calendarDate;
+      this._selectedDates.push(calendarDate);
+    } else if (this._selectedRangeDates.startDate && !this._selectedRangeDates.endDate) {
+      if (calendarDate.getTime() > this._selectedRangeDates.startDate.getTime()) {
+        this._selectedRangeDates.endDate = calendarDate;
+        this._selectedDates.push(calendarDate);
+      } else if (calendarDate.getTime() < this._selectedRangeDates.startDate.getTime()) {
+        const temp = this._selectedRangeDates.startDate;
+
+        this._selectedRangeDates.startDate = calendarDate;
+        this._selectedRangeDates.endDate = temp;
+        this._selectedDates.splice(
+          0,
+          this._selectedDates.length,
+          this._selectedRangeDates.startDate,
+          this._selectedRangeDates.endDate
+        );
+      }
+    } else if (this._selectedRangeDates.startDate && this._selectedRangeDates.endDate) {
+      this._selectedRangeDates.startDate = calendarDate;
+      this._selectedRangeDates.endDate = undefined;
+      this._selectedDates.splice(0, this._selectedDates.length, this._selectedRangeDates.startDate);
+    }
+    this.setHoverClass();
+  }
```

Comment:
This has excessive cognitive complexity, maybe this could be simpler?

Reply:
i dont believe there is , there are four case in there.First if there is not any startDay or endDay,it sets given date as start date.if there is any start date,then there are 2 options.Given date can be bigger than start date,then given date will be end date,if given date smaller than the star date,it should be start date and temp start date should be end date(and start and end date pushed selected dates array).If there are both start date and end date,then they should cleared and start date should be given date